src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: string): GroupsMap<T> {
+  return items.reduce(
+    (hash: GroupsMap<T>, obj: T) => ({
+      ...hash,
+      [obj[key]]: (hash[obj[key]] || []).concat(obj)
+    }),
+    /*  eslint comma-dangle: ["error", "never"] */
```

Comment:
Why did you add this eslint rule here?
Instead it just add comma where eslint ask)

Reply:
Ok, done)